include/libopencm3/stm32/l4/flash.h

```diff
@@ -0,0 +1,256 @@
+ *
+ * @version 1.0.0
+ *
+ * @author @htmlonly &copy; @endhtmlonly 2016 Benjamin Levine <benjaminmileslevine@gmail.com>
+ *
+ * @date 12 February 2016
+ *
+ * LGPL License Terms @ref lgpl_license
+ *  */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2016 Benjamin Levine <benjaminmileslevine@gmail.com>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * For details see:
+ * RM0351 Reference manual: STM32L4x6 advanced ARM®-based 32-bit MCUs
+ * December 2015, Doc ID 024597 Rev 3
+ */
+
+/**@{*/
+#ifndef LIBOPENCM3_FLASH_H
+#define LIBOPENCM3_FLASH_H
+
+/* --- FLASH registers ----------------------------------------------------- */
+
+#define FLASH_ACR			MMIO32(FLASH_MEM_INTERFACE_BASE + 0x00)
+#define FLASH_PDKEYR		MMIO32(FLASH_MEM_INTERFACE_BASE + 0x04)
+#define FLASH_KEYR			MMIO32(FLASH_MEM_INTERFACE_BASE + 0x08)
+#define FLASH_OPTKEYR		MMIO32(FLASH_MEM_INTERFACE_BASE + 0x0C)
+#define FLASH_SR			MMIO32(FLASH_MEM_INTERFACE_BASE + 0x10)
+#define FLASH_CR			MMIO32(FLASH_MEM_INTERFACE_BASE + 0x14)
+#define FLASH_ECCR          MMIO32(FLASH_MEM_INTERFACE_BASE + 0x18)
+#define FLASH_OPTR			MMIO32(FLASH_MEM_INTERFACE_BASE + 0x20)
+#define FLASH_PCROP1SR      MMIO32(FLASH_MEM_INTERFACE_BASE + 0x24)
+#define FLASH_PCROP1ER      MMIO32(FLASH_MEM_INTERFACE_BASE + 0x28)
+#define FLASH_WRP1AR        MMIO32(FLASH_MEM_INTERFACE_BASE + 0x2C)
+#define FLASH_WRP1BR        MMIO32(FLASH_MEM_INTERFACE_BASE + 0x30)
+#define FLASH_PCROP2SR      MMIO32(FLASH_MEM_INTERFACE_BASE + 0x44)
+#define FLASH_PCROP2ER      MMIO32(FLASH_MEM_INTERFACE_BASE + 0x48)
+#define FLASH_WRP2AR        MMIO32(FLASH_MEM_INTERFACE_BASE + 0x4C)
+#define FLASH_WRP2BR        MMIO32(FLASH_MEM_INTERFACE_BASE + 0x50)
+
+/* --- FLASH_ACR values ---------------------------------------------------- */
+
+#define FLASH_ACR_SLEEP_PD			(1 << 14)
+#define FLASH_ACR_RUN_PD			(1 << 13)
+#define FLASH_ACR_DCRST			    (1 << 12)
+#define FLASH_ACR_ICRST			    (1 << 11)
+#define FLASH_ACR_DCEN			    (1 << 10)
+#define FLASH_ACR_ICEN			    (1 << 9)
+#define FLASH_ACR_PRFTEN		    (1 << 8)
+
+#define FLASH_ACR_LATENCY_SHIFT     0
+#define FLASH_ACR_LATENCY_MASK      0x03
+
+#define FLASH_ACR_LATENCY_0WS		0x00
+#define FLASH_ACR_LATENCY_1WS		0x01
+#define FLASH_ACR_LATENCY_2WS		0x02
+#define FLASH_ACR_LATENCY_3WS		0x03
```

Comment:
I think a `FLASH_ACR_LATENCY_4WS` is needed.
See in 3.7.1 in L4x6 reference manual ([RM0351](http://www.st.com/resource/en/reference_manual/dm00083560.pdf)).